### PR TITLE
Add client for ocr validation endpoint

### PR DESCRIPTION
**Pull request by rkondratowicz**
https://tools.hmcts.net/jira/browse/BPS-685

Not used (yet), but with tests.

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/ocrvalidation/OcrValidationClientTest.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+@ActiveProfiles( {
+    IntegrationContextInitializer.PROFILE_WIREMOCK,
+    Profiles.SERVICE_BUS_STUB,
+    Profiles.STORAGE_STUB
+})
+@AutoConfigureWireMock
+@IntegrationTest
+@RunWith(SpringRunner.class)
+public class OcrValidationClientTest {
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/ocrvalidation/OcrValidationClientTest.java]: Could we please add tests when there is success and when there are only warnings ? and also for invalid s2s token and 403 scenarios ?

Comment (rkondratowicz) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/ocrvalidation/OcrValidationClientTest.java]: done

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/ocrvalidation/client/OcrValidationClient.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+@Component
+public class OcrValidationClient {
+
+    private final RestTemplate restTemplate;
+
+    public OcrValidationClient(RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    public ValidationResponse validate(String url, FormData formData, String s2sToken) {
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/ocrvalidation/client/OcrValidationClient.java]: Could it get some s2s token provider maybe?

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/ocrvalidation/client/model/req/OcrDataField.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.ocrvalidation.client.model.req;
+
+public class OcrDataField {
+
+    public final String name;
+    public final String value;
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/ocrvalidation/client/model/req/OcrDataField.java]: Don't these two need `@JsonProperty` annotation?

**Comment by lgonczar** [comment]
Worth adding some wiremock tests or whatever kind of tests that would check if the client is actually hitting some enpoint.

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/ocrvalidation/client/OcrValidationClient.java (hunk 2 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/ocrvalidation/client/OcrValidationClient.java]: my plan is for the service to take a provider in constructor

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/ocrvalidation/client/model/req/OcrDataField.java (hunk 3 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/ocrvalidation/client/model/req/OcrDataField.java]: only if you want to set the json property name explicitly

**Comment by lgonczar** [comment]
> Worth adding some wiremock tests or whatever kind of tests that would check if the client is actually hitting some endpoint.

Which is exactly what we've already got. Sorry, my bad.

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/ocrvalidation/client/OcrValidationClient.java (hunk 2 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/ocrvalidation/client/OcrValidationClient.java]: to be more specific, I plan to create `OcrValidationService` that will take envelope input models as parameters, map them to REST api models and call the client from this PR.

**Pull request merged**